docs: Document GiST and SP-GiST geography indexes (#661)
serverpod/serverpod_docs: docs/06-concepts/06-database/04-indexing.md

    type: gist
```

Use `spgist` by setting the index `type` explicitly:

```yaml
class: DeliveryZone
table: delivery_zone
fields:
  name: String
  boundary: GeographyPolygon
indexes:
  delivery_zone_boundary_idx:
    fields: boundary
    type: spgist
```

:::tip
A spatial index accelerates all spatial operations (`intersects`, `distanceWithin`, `distance`, `contains`, `within`). For tables with many rows and frequent spatial queries, adding one is strongly recommended.
:::

> Geography fields only support the `gist` and `spgist` index types. Specifying any other type fails code generation with `The "type" property must be one of: gist, spgist.` An index may cover several geography columns, but geography and non-geography fields cannot be mixed in the same index — doing so fails with `Mixing geography and non-geography fields in the same index is not allowed.`

Reviewer: Nice catch on the `dwithin` → `distanceWithin` fix, and the error strings match the source. One small thing: we keep em dashes out of the docs. Splitting this into two sentences fixes it (and reads a touch cleaner):

```suggestion
Geography fields only support the `gist` and `spgist` index types. Specifying any other type fails code generation with `The "type" property must be one of: gist, spgist.` An index may cover several geography columns, but it cannot mix geography and non-geography fields. Doing so fails with `Mixing geography and non-geography fields in the same index is not allowed.`
```

Author: Good catch, dropped the em dash and split it into two sentences per your suggestion. Pushed in 56d38b9.

Author: This is fixed. Thank you

Author: @developerjamiu, kindly review.
